google/cloud/pubsub/internal/streaming_subscription_batch_source.cc

```diff
@@ -323,16 +331,22 @@ void StreamingSubscriptionBatchSource::OnFinish(Status status) {
 
 void StreamingSubscriptionBatchSource::DrainQueues(
     std::unique_lock<std::mutex> lk, bool /*force_flush*/) {
-  if (deadlines_queue_.empty()) return;
+  if (deadlines_queue_.empty() && !needs_stream_ack_deadline_update_) return;
   if (stream_state_ != StreamState::kActive || pending_write_) return;
   auto stream = stream_;
   pending_write_ = true;
 
+  auto stream_ack_deadline = std::int32_t{0};
```

Comment:
Opt: It would be nicer, I reckon, to make `stream_ack_deadline` a `std::chrono::seconds`, and leave the `.count()` to the same expression as the `set_stream_ack_deadline_seconds()`.  The longer we stay in the `chrono` domain, the better.

[Aside: It is a shame that the proto field is called "deadline" (which generally implies a time) instead of "timeout" (which generally implies a duration).  And I guess we might as well mirror their naming.]